x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsIntegTests.java

```diff
@@ -540,6 +546,42 @@ public void testCreateAndRestorePartialSearchableSnapshot() throws Exception {
         final RestoreSnapshotResponse restoreSnapshotResponse = client().execute(MountSearchableSnapshotAction.INSTANCE, req).get();
         assertThat(restoreSnapshotResponse.getRestoreInfo().failedShards(), equalTo(0));
 
+        final Map<Integer, SnapshotIndexShardStatus> snapshotShards = clusterAdmin().prepareSnapshotStatus(fsRepoName)
+            .setSnapshots(snapshotInfo.snapshotId().getName())
+            .get()
+            .getSnapshots()
+            .get(0)
+            .getIndices()
+            .get(indexName)
+            .getShards();
+
+        final IndicesStatsResponse indicesStatsResponse = client().admin()
+            .indices()
+            .prepareStats(restoredIndexName)
+            .clear()
+            .setStore(true)
+            .get();
+        assertThat(indicesStatsResponse.getShards().length, greaterThan(0));
+        long totalExpectedSize = 0;
+        for (ShardStats shardStats : indicesStatsResponse.getShards()) {
+            StoreStats store = shardStats.getStats().getStore();
+            assertThat(shardStats.getShardRouting().toString(), store.getReservedSize().getBytes(), equalTo(0L));
+            assertThat(shardStats.getShardRouting().toString(), store.getLocalSize().getBytes(), equalTo(0L));
+
+            // the extra segments file created for bootstrap new history and associate translog makes us unable to precisely assert this.
+            final long expectedSize = snapshotShards.get(shardStats.getShardRouting().getId()).getStats().getTotalSize();
+            assertThat(shardStats.getShardRouting().toString(), store.getSize().getBytes(), greaterThanOrEqualTo(expectedSize));
+            // we expect the new segments file to make it significantly less than 2x.
+            assertThat(shardStats.getShardRouting().toString(), store.getSize().getBytes(), lessThan(expectedSize * 2));
```

Comment:
This seems like it might be flaky - why `*2`? The index could in theory have one tiny doc per segment, are we sure that each segment will always be larger than the per-segment entry in the `segments_N` file?

If you're sure this is ok, can we explain the reasoning in a more detailed comment?

Reply:
The 2x is there because there is one `segments_N` file already and the bootstrap/translog association adds another `segments_N` file with only changes being the two UUIDs that are now different. AFAICS, that will always lead to a new `segments_N` file of identical size. Thus a worst case is 2x. I added a comment to clarify this in the code too.

Reply:
Ah good point I had forgotten about the older `segments_N` file. Yes this looks solid now.